Ice/MenuBar/Appearance/MenuBarOverlayPanel.swift

```diff
@@ -328,6 +324,12 @@ final class MenuBarOverlayPanel: NSPanel {
             return
         }
 
+        // Validate before showing to ensure panel should be visible on this screen.
```

Comment:
This ensures the panel is validated before being shown. If it is in full-screen mode, it should be hidden.